guides/source/sign_up.md

```diff
@@ -0,0 +1,1652 @@
+**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON <https://guides.rubyonrails.org>.**
+
+Sign Up and Registration
+========================
+
+This guide covers adding Sign Up and Registration to the store e-commerce application in the [Getting Started Guide](getting_started.html)). We will use the final code from that guide as a starting place.
+
+After reading this guide, you will know:
+* How to add Sign Up
+* How to rate limit controller actions
+* How to create a nested layout
+* How to separate controllers by role
+* How to write tests for users with different roles
+
+--------------------------------------------------------------------------------
+
+Introduction
+------------
+
+One of the most common features to add to any application is a sign up process for registering new users. The e-commerce application we've built so far only has authentication and users must be created in the Rails console or a script.
+
+This feature is a required foundation for adding other features. For example, to let users create wishlists, they will need to be able to sign up first before they can create a wishlist associated with their account.
+
+Let's get started!
+
+Adding Sign Up
+--------------
+
+We've already used the Rails authentication generator to allow users to login to their accounts. The generator created a `User` model with `email_address:string` and `password_digest:string` columns in the database. It also added `has_secure_password` to the `User` model which handles passwords and confirmations. This takes care of most of what we need to add sign up to our application.
+
+### Adding Names To Users
+
+One other feature we probably want for users is to collect their name on sign up. Let's start by adding `first_name` and `last_name` columns to the database.
```

Comment:
It's also a good idea to collect the user's name at sign up. This allows us to personalize their experience and address them directly in the view. Let's start by adding `first_name` and `last_name` columns to the database.